internal/storage/bucket_handle_test.go

```diff
@@ -187,7 +187,19 @@ func (t *BucketHandleTest) TestDeleteObjectMethodWithMissingObject() {
 			MetaGenerationPrecondition: nil,
 		})
 
-	AssertEq("storage: object doesn't exist", err.Error())
+	AssertEq("gcs.NotFoundError: storage: object doesn't exist", err.Error())
+}
+
+func (t *BucketHandleTest) TestDeleteObjectMethodWithMissingGeneration() {
```

Comment:
add a test case for explicitly passing generation as zero.

Reply:
I thought of it but the fake storage serve doesn't respect that: https://github.com/GoogleCloudPlatform/gcsfuse/blob/bb041a9ee55fd338fe134b49a9fdfd2a92852a7c/internal/storage/bucket_handle_test.go#L40. 

But I agree that we should still add it. Please let me know if you think otherwise.

Reply:
so missing generation is not a valid test case then?

Reply:
Missing generation as a unit test is valid to ensure that the DeleteObject doesn't fail with missing generation (and it not only specific to this change). 
From an idempotency check point of view, it is not valid.